### PR TITLE
operator: cache operator influence

### LINKED ISSUE #6235: operator: cache influence
In past, the influence of the operator maybe not same in the life of the operator. The potential cause 
1. the region size maybe changed
2. the region has been removed(merged)

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

Issue Number: Close #6235

### What is changed and how does it work?

```commit-message

```

### Check List

Tests

- Unit test

Code changes

Side effects

Related changes

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: pkg/core/storelimit/limit.go [hunk 1]
```diff
@@ -33,11 +33,21 @@ const (
...
+// 4. ack will put back the cost into the limit for the next waiting operator after the operator is finished.
+// the cost is the operator influence, so the influence should be same in the life of the operator.
 type StoreLimit interface {
 	// Available returns true if the store can accept the operator
 	Available(cost int64, typ Type, level constant.PriorityLevel) bool
 	// Take takes the cost of the operator, it returns false if the store can't accept any operators.
 	Take(count int64, typ Type, level constant.PriorityLevel) bool
 	// Reset resets the store limit
 	Reset(rate float64, typ Type)
+	// Ack put back the cost into the limit for the next waiting operator after the operator is finished.
```
Comment (rleungx) [inline pkg/core/storelimit/limit.go:51]: Suggest adding comments to explain this kind of interface is only used by snapshot limit.

File: pkg/schedule/operator/influence.go [hunk 2]
```diff
@@ -52,8 +59,18 @@ type StoreInfluence struct {
 	StepCost     map[storelimit.Type]int64
 }
 
+func (s *StoreInfluence) add(other StoreInfluence) {
+	s.RegionCount += other.RegionCount
+	s.RegionSize += other.RegionSize
+	s.LeaderSize += other.LeaderSize
+	s.LeaderCount += other.LeaderCount
```
Comment (rleungx) [inline pkg/schedule/operator/influence.go:66]: Do we need to consider the witness?

File: pkg/schedule/operator/influence.go [hunk 3]
```diff
@@ -52,8 +59,18 @@ type StoreInfluence struct {
 	StepCost     map[storelimit.Type]int64
 }
 
+func (s *StoreInfluence) add(other StoreInfluence) {
```
Comment (CabinfeverB) [inline pkg/schedule/operator/influence.go:62]: Can we use pointer?

File: pkg/schedule/operator_controller.go [hunk 4]
```diff
@@ -711,20 +726,18 @@ func (oc *OperatorController) OperatorCount(kind operator.OpKind) uint64 {
...
-	}
+	influence := operator.NewOpInfluence()
 	oc.RLock()
 	defer oc.RUnlock()
 	for _, op := range oc.operators {
 		if !op.CheckTimeout() && !op.CheckSuccess() {
 			region := cluster.GetRegion(op.RegionID())
 			if region != nil {
-				op.UnfinishedInfluence(influence, region)
+				op.UnfinishedInfluence(*influence, region)
```
Comment (CabinfeverB) [inline pkg/schedule/operator_controller.go:736]: Can we change the `opInfluence` as pointer in `UnfinishedInfluence`？

**Review by CabinfeverB — COMMENTED** [review]
Most LGTM

File: pkg/schedule/operator/influence.go (hunk 2 above)
Comment (bufferflies) [reply pkg/schedule/operator/influence.go:66]: good catch，I will add it

File: pkg/schedule/operator/influence.go (hunk 3 above)
Comment (bufferflies) [reply pkg/schedule/operator/influence.go]: yes

File: pkg/core/storelimit/limit.go (hunk 1 above)
Comment (bufferflies) [reply pkg/core/storelimit/limit.go:51]: fixed it

File: pkg/schedule/operator_controller.go (hunk 4 above)
Comment (bufferflies) [reply pkg/schedule/operator_controller.go]: I will revert this change

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**